src/providers/clojure.rs

```diff
@@ -0,0 +1,167 @@
+use super::Provider;
+use crate::nixpacks::{
+    app::App,
+    environment::Environment,
+    nix::pkg::Pkg,
+    phase::{BuildPhase, SetupPhase, StartPhase},
+};
+use anyhow::Result;
+use regex::Regex;
+
+const DEFAULT_JDK_PKG_NAME: &'static &str = &"jdk8";
+pub struct ClojureProvider {}
+
+impl Provider for ClojureProvider {
+    fn name(&self) -> &str {
+        "clojure"
+    }
+
+    fn detect(&self, app: &App, _env: &Environment) -> Result<bool> {
+        Ok(app.includes_file("project.clj"))
+    }
+
+    fn setup(&self, _app: &App, _env: &Environment) -> Result<Option<SetupPhase>> {
+        Ok(Some(SetupPhase::new(vec![
+            Pkg::new("leiningen"),
+            Pkg::new("jdk8"),
+        ])))
+    }
+
+    fn build(&self, _app: &App, _env: &Environment) -> Result<Option<BuildPhase>> {
+        Ok(Some(BuildPhase::new("lein uberjar".to_string())))
+    }
+
+    fn start(&self, _app: &App, _env: &Environment) -> Result<Option<StartPhase>> {
+        Ok(Some(StartPhase::new(
+            "java $JAVA_OPTS -jar target/uberjar/*standalone.jar".to_string(),
+        )))
+    }
+}
+
+impl ClojureProvider {
+    pub fn get_nix_jdk_package(app: &App, env: &Environment) -> Result<Pkg> {
+        // Fetch version from configs
+        let mut custom_version = env.get_config_variable("JDK_VERSION");
+
+        // If not from configs, get it from the .jdk-version file
+        if custom_version.is_none() && app.includes_file(".jdk-version") {
+            custom_version = Some(app.read_file(".jdk-version")?);
+        }
+
+        // If it's still none, return default
+        if custom_version.is_none() {
+            return Ok(Pkg::new(DEFAULT_JDK_PKG_NAME));
+        }
+        let custom_version = custom_version.unwrap();
```

Comment:
Nit: This can be shortened to

```rs
        let custom_version = match custom_version {
            Some(v) => v,
            None => return Ok(Pkg::new(DEFAULT_JDK_PKG_NAME)),
        };
```

It also reads a bit nicer and doesn't have to unwrap